miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java

```diff
@@ -116,16 +121,46 @@ private static SampleDto writeUrls(SampleDto sampleDto, UriComponentsBuilder uri
 
   @RequestMapping(value = "/samples", method = RequestMethod.GET, produces = { "application/json" })
   @ResponseBody
-  public ResponseEntity<Set<SampleDto>> getSamples(UriComponentsBuilder uriBuilder) throws IOException {
-    Set<Sample> samples = sampleService.getAll();
-    if (samples.isEmpty()) {
-      throw new RestException("No samples found", Status.NOT_FOUND);
-    } else {
-      Set<SampleDto> sampleDtos = Dtos.asSampleDtos(samples);
+  public ResponseEntity<?> getSamples(HttpServletRequest request, HttpServletResponse response, UriComponentsBuilder uriBuilder)
```

Comment:
Should return either `ResponseEntity<DataTablesResponseDto<SampleDto>>` or just `DataTablesResponseDto<SampleDto>`. Probably prefer the latter since the ResponseEntity isn't really doing anything. (`200 OK` is default response status)